### ISSUE TITLE
Interpolation of matches does not work within quotes

### ISSUE THREAD

**Issue by damiendr** [kmsquire/Match.jl#22]
*Note: I'm new to Julia so maybe I'm missing some subtlety here.*

It seems that interpolation within quoted expressions does not work directly in the match block:
```julia
arr = [1, 2]
@match arr begin
    [a, b] => :($a + $b)
end
```
```
LoadError: UndefVarError: a not defined
```

But it does when wrapped in a function call:
```julia
expr(a, b) = :($a + $b)
@match arr begin
    [a, b] => expr(a,b)
end
```
```
:(1 + 2)
```

**Comment by kmsquire** [comment]
Thanks for the report. I'm honesty not sure how easy it would be to implement this--it might mean just another test for expressions with substitution variables. Or, it might not be possible without reimplementing the Julia parser (or using JuliaParser.jl).  

Unfortunately, my Julia dev time has been rather limited recently, so I'm not sure when I'll be able to look into this. If it's important to you to have it sooner rather than later, I would suggest trying to work through the source and posting questions back here. (It might be a bit dense--if so, sorry!)

**Comment by damiendr** [comment]
Submitted a tentative pull request — I think it needs checking for potential regressions, even though all tests pass.

**Comment by kmsquire** [comment]
Fixed by #23
